public/mobile-app/src/routes/page.svelte.test.ts

```diff
@@ -1,11 +1,71 @@
-import { describe, test, expect } from 'vitest'
+import { describe, test, expect, vi } from 'vitest'
 import '@testing-library/jest-dom/vitest'
-import { render, screen } from '@testing-library/svelte'
+import { render, screen, waitFor } from '@testing-library/svelte'
 import Page from './+page.svelte'
 
 describe('/+page.svelte', () => {
-  test('should render h1', () => {
+  test('should render h1 without user email when email and pushSubscription are not set', () => {
+    // When
     render(Page)
-    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
+
+    // Then
+    const title = screen.getByRole('heading', { level: 1 })
+    expect(title).toHaveTextContent("Bienvenue sur l'application AMI")
+  })
+
+  test('should render h1 with user email when email and pushSubscription are set', () => {
+    // Given
+    window.localStorage.setItem('emailLocalStorage', 'test@email.fr')
+    window.localStorage.setItem('pushSubscriptionLocalStorage', '{}')
+
+    // When
+    render(Page)
+
+    // Then
+    const title = screen.getByRole('heading', { level: 1 })
+    expect(title).toHaveTextContent("Bienvenue test@email.fr sur l'application AMI")
+  })
+
+  test('should display authentication status message when user clicks on authentication button and authentication works', async () => {
+    // Given
+    globalThis.Notification = {
+      requestPermission: () => true,
+      permission: 'granted',
+    }
+    const registration = {
+      pushManager: {
+        subscribe: vi.fn(() => Promise.resolve('fake pushSubscription')),
+      },
+    }
+    globalThis.navigator = {
+      serviceWorker: {
+        ready: new Promise((resolve) => {
+          resolve(registration)
+        }),
+      },
+    }
```

Comment:
That's some clever setup, nice, I wonder if vitest offers some kind of fixtures/dependency injection.
In any case, we might want to move those helpers in another file, to be reused in future tests.

We could also postpone that until it's needed, of course ;)